x-pack/plugins/observability_solution/dataset_quality/common/utils/quality_helpers.ts

```diff
@@ -8,10 +8,14 @@
 import { POOR_QUALITY_MINIMUM_PERCENTAGE, DEGRADED_QUALITY_MINIMUM_PERCENTAGE } from '../constants';
 import { QualityIndicators } from '../types';
 
-export const mapPercentageToQuality = (percentage: number): QualityIndicators => {
-  return percentage > POOR_QUALITY_MINIMUM_PERCENTAGE
-    ? 'poor'
-    : percentage > DEGRADED_QUALITY_MINIMUM_PERCENTAGE
-    ? 'degraded'
-    : 'good';
+export const mapPercentageToQuality = (percentages: number[]): QualityIndicators => {
```

Comment:
> [!Note]
> Quality is now calculated taking into account multiple percentages. The quality status will be given by the most restricting status in all.

Reply:
`some` provides an early termination vs the for loop, that's why I preferred to use it here